DSCResources/MSFT_xDNSServerAddress/MSFT_xDNSServerAddress.psm1

```diff
@@ -80,93 +125,124 @@ function Test-TargetResource
         [String]$InterfaceAlias,
 
         [Parameter(Mandatory)]
-        [ValidateSet("IPv4", "IPv6")]
+        [ValidateSet('IPv4', 'IPv6')]
         [String]$AddressFamily
     )
+    # Flag to signal whether settings are correct
+    [Boolean]$requiresChanges = $false
 
-    Test-Properties @PSBoundParameters
-}
+    try
+    {        
+        Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
+            'Checking the DNS Server Address ...' 
+            ) -join '' )
+
+        #Validate the Settings passed
+        Foreach ($ServerAddress in $Address) {       
+            Test-ResourceProperty `
+                -Address $ServerAddress `
+                -AddressFamily $AddressFamily `
+                -InterfaceAlias $InterfaceAlias
+        }
 
+        #Get the current DNS Server Addresses based on the parameters given.
+        $currentAddress = (Get-DnsClientServerAddress `
+            -InterfaceAlias $InterfaceAlias `
+            -AddressFamily $AddressFamily `
+            -ErrorAction Stop).ServerAddresses
+
+        #Check if the Server addresses are the same as the desired addresses.
+        if(@(Compare-Object `
+            -ReferenceObject $currentAddress `
+            -DifferenceObject $Address `
+            -SyncWindow 0).Length -gt 0)
+        {
+            $requiresChanges = $true
+            Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
+                "DNS Servers are not correct. Expected $Address, actual $currentAddress." 
+                ) -join '' )
+        }
+        else 
+        { 
+            #Test will return true in this case
+            Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
+                'DNS Servers are set correctly.' 
+                ) -join '' )
+        }
+    }
+    catch
```

Comment:
Doesn't this swallow any exception you threw in Test-ResourceProperty considering the throw (in Test-ResourceProperty) is happening within your try/catch on this level?

Given that the helper function is expected to throw do we really need the try/catch here?